### PR TITLE
C402349: Create custom title when session locale is not "English" (spitfire) (TaaS)

### PR DESCRIPTION
**Pull request by IlyaLyskavets**
[Test Rail](https://foliotest.testrail.io/index.php?/cases/view/402349)

Allure: image
Tests that could have been affected:
image

### CODE REVIEW

File: cypress/support/fragments/eholdings/eHoldingsTitles.js [hunk 1]
```diff
@@ -28,9 +28,9 @@ export default {
       eHoldingsTitle.waitLoading(title);
     });
   },
-  create: (packageName, titleName = `title_${getRandomPostfix()}`) => {
-    cy.do(resultSection.find(Button('New')).click());
-    eHoldingsNewCustomTitle.waitLoading();
+  create: (packageName, titleName = `title_${getRandomPostfix()}`, paneTitle) => {
+    cy.do(resultSection.find(Button({ href: '/eholdings/titles/new' })).click());
+    eHoldingsNewCustomTitle.waitLoading(paneTitle);
```
Comment (dnovikau) [inline cypress/support/fragments/eholdings/eHoldingsTitles.js:33]: This change could affect tests where the method is already used. Please include execution of all tests where the method is used to result report or remove this change.

File: cypress/support/fragments/eholdings/eHoldingsTitles.js [hunk 2]
```diff
@@ -28,9 +28,9 @@ export default {
       eHoldingsTitle.waitLoading(title);
     });
   },
-  create: (packageName, titleName = `title_${getRandomPostfix()}`) => {
-    cy.do(resultSection.find(Button('New')).click());
-    eHoldingsNewCustomTitle.waitLoading();
+  create: (packageName, titleName = `title_${getRandomPostfix()}`, paneTitle) => {
+    cy.do(resultSection.find(Button({ href: '/eholdings/titles/new' })).click());
```
Comment (dnovikau) [inline cypress/support/fragments/eholdings/eHoldingsTitles.js:32]: Is this change required? In some cases it's preferable to use previous aproach

File: cypress/support/fragments/eholdings/eHoldingsPackages.js [hunk 3]
```diff
@@ -109,7 +109,7 @@ export default {
   getCustomPackageViaApi: () => {
     cy.okapiRequest({
       path: 'eholdings/packages',
-      searchParams: { 'filter[custom]': true, count: 10, pageSize: 10 },
+      searchParams: { 'filter[custom]': true /* count: 10, pageSize: 10 */ },
```
Comment (dnovikau) [inline cypress/support/fragments/eholdings/eHoldingsPackages.js:112]: Please remove commented code and leave comment why parameters were removed.

File: cypress/support/fragments/eholdings/eHoldingsTitles.js (hunk 1 above)
Comment (IlyaLyskavets) [reply cypress/support/fragments/eholdings/eHoldingsTitles.js:33]: It was one test where this method is used. Added successful run in the description of the pull request

File: cypress/support/fragments/eholdings/eHoldingsPackages.js (hunk 3 above)
Comment (IlyaLyskavets) [reply cypress/support/fragments/eholdings/eHoldingsPackages.js]: removed and added explaining comment

**Review by dnovikau — APPROVED** [review]

File: cypress/support/fragments/eholdings/eHoldingsTitles.js (hunk 2 above)
Comment (IlyaLyskavets) [reply cypress/support/fragments/eholdings/eHoldingsTitles.js:32]: @dnovikau I just changed it back to not overcomplicate things. But I had to delete eHoldingsNewCustomTitle.waitLoading() step from the method completely since inside it checks name of the Pain (in English), but in flow on the test this title won't be in English. So that was the point of previous overcomplications, so the question what is better to have: deleted waitLoading step or one more optional parameter with name of the Pane?

Comment (dnovikau) [reply cypress/support/fragments/eholdings/eHoldingsTitles.js:32]: From localization teststing perspective approach with locator using attribute values seams more convenient. Please leave proposed realization:
```JS
cy.do(resultSection.find(Button({ href: '/eholdings/titles/new' })).click());
```

**Review by zentestuken — APPROVED** [review]
